sys/xtimer/Kconfig

```diff
@@ -9,9 +9,7 @@ menuconfig MODULE_XTIMER
     bool "xtimer"
     depends on HAS_PERIPH_TIMER
     depends on TEST_KCONFIG
-
-    # use timer peripheral unless ztimer compatibility module is used
-    select MODULE_PERIPH_TIMER if HAS_PERIPH_TIMER && !MODULE_XTIMER_ON_ZTIMER && !MODULE_ZTIMER_XTIMER_COMPAT
+    select MODULE_PERIPH_TIMER if HAS_PERIPH_TIMER
```

Comment:
I think the idea of this was not to force the timer backend when compatibility is in place, leaving the option of the hardware to the Ztimer config. Why does this change?

Reply:
will split out in another commit if justfied

Reply:
Hmm well all green

Reply:
@leandrolanzieri does everything else Kconfig related look good to you?